tsconfig.base.json

```diff
@@ -1,31 +1,37 @@
 {
   "compilerOptions": {
-    "allowSyntheticDefaultImports": true,
-    "allowUnreachableCode": false,
-    "allowUnusedLabels": false,
-    "esModuleInterop": true,
-    "exactOptionalPropertyTypes": true,
-    "forceConsistentCasingInFileNames": true,
-    "importHelpers": false,
+    "target": "ESNext",
     "lib": [
-      "ESNext",
+      "ESNext"
     ],
     "module": "ESNext",
     "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "importHelpers": false,
+    "importsNotUsedAsValues": "error",
     "newLine": "LF",
     "noEmitOnError": true,
-    "noImplicitReturns": true,
-    "noUnusedLocals": false,
-    "noUnusedParameters": false,
-    "pretty": true,
-    "resolveJsonModule": true,
-    "sourceMap": false,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
     "strict": true,
+    "noImplicitAny": true,
+    "strictNullChecks": true,
+    "strictFunctionTypes": true,
+    "strictBindCallApply": true,
+    "strictPropertyInitialization": true,
+    "noImplicitThis": true,
+    "useUnknownInCatchVariables": true,
     "alwaysStrict": true,
-    "target": "ES2018",
+    "exactOptionalPropertyTypes": true,
+    "noImplicitReturns": true,
+    // "noUncheckedIndexedAccess": true,
```

Comment:
this isn't valid JSON

Reply:
tsconfig files are read as `jsonc` rather than `json`. You should be able to configure your editor to parse the files as such.